Feature/search improvement

mpuckett159 opened the pull request:
Closes #20

Two main commits here, one to enable query params, second to cut the HTML code over to make use of them.

mpuckett159 committed:
Allow query parameters

app.py:

 * imports
Add request, redirect, url_for from flask library to support query parameter usage and redirecting from the root page to the license page.

 * /
Change behavior of root to redirect to license page.

 * /license
Add get for parameter, then check if the parameter was passed and load the page if so. If not, load the default HTML.

 * /badge
Same changes as license.

 * /name
Same changes as license.

This change is adding the ability for users to use query parameters if they want. Next commit will be to update the HTML to utilize these new changes.

mpuckett159 committed:
Fix license change issue

Copy/paste doh. Change the license Flask app to use the license_lookup function instead of the badge_lookup function.

mpuckett159 committed:
Swap to query params

This commit cuts over the website from the URL parsing format to utilize query parameters.

app.py:
 * LICENSE_CONTEXT
Add query_param to render HTML form properly. Change lookup_url to redirect back to the same url.

 * BADGE_CONTEXT
Same changes as LICENSE_CONTEXT

 * NAME_CONTEXT
Same changes as LICENSE_CONTEXT.

index.html:
Changing root form to be explicitly a license form.
Change form actions to execute GET request with query parameters as defined in the various context dicts.
Remove jquery stuff because it isn't needed anymore.

AetherUnbound commented on src/views/index.html:
```diff
@@ -46,21 +46,5 @@ <h1>
 
-
-<!-- Your web-app is https, so your scripts need to be too -->
-<script src="https://code.jquery.com/jquery-2.2.1.min.js"
```
Yay, less javascript!

AetherUnbound commented on src/app.py:
```diff
@@ -16,13 +25,21 @@
-    """Displays the homepage."""
+    """License lookup form render"""
+    # Check for badge query params and load if it exists.
+    license = request.args.get('license')
+    if license:
+        html = dataset.license_lookup(license)
+        return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
+
+    # Return the basic lookup form if not
     return render_template("index.html", **LICENSE_CONTEXT)
```
For each of these cases, I'd suggest the following change for simplicity:

```suggestion
    license = request.args.get('license')
    html = ""
    if license:
        html = dataset.license_lookup(license)

    return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
```

This only hydrates the extra HTML if it's available, and still renders everything well when it isn't.

mpuckett159 replied:
Oh this makes sense, I didn't think about that. I'll make those changes.

AetherUnbound reviewed (commented):
Dang, this is awesome! You're right, way easier than I was expecting. I checked that the old links are still preserved and they are! One comment about formatting, then I think this is good to go :rocket:

mpuckett159 committed:
Simplify page renders

Simplifying the HTML rendering for the Flask app. This removes some redundancy.

AetherUnbound reviewed (approved):
Way to go :sunglasses:
